src/bentoml/_internal/runner/strategy.py

```diff
@@ -121,15 +124,35 @@ def get_worker_env(
         environ: dict[str, t.Any] = {}
         if resource_request is None:
             resource_request = system_resources()
-
         # use nvidia gpu
-        nvidia_gpus = get_resource(resource_request, "nvidia.com/gpu")
+        nvidia_gpus: list[int] | None = get_resource(resource_request, "nvidia.com/gpu")
         if (
             nvidia_gpus is not None
             and len(nvidia_gpus) > 0
             and "nvidia.com/gpu" in runnable_class.SUPPORTED_RESOURCES
         ):
-            dev = str(nvidia_gpus[worker_index // workers_per_resource])
+            if isinstance(workers_per_resource, float):
+                # NOTE: We hit this branch when workers_per_resource is set to
+                # float, for example 0.5 or 0.25
+                assigned_resource_per_worker = int(1 / workers_per_resource)
```

Comment:
should we check if workers_per_resource < 1 when it's float?

Reply:
This PR actually just assumes it is <1. I think it makes sense to only hit this branch when it is < 1, since Idk how would you do it with something like 2.5 workers.

Reply:
ye I mean throw an error if it's > 1?

Reply:
I see you add the logic. gj!